lib/code_fund_web/controllers/ad_serve_controller.ex

```diff
@@ -15,31 +15,39 @@ defmodule CodeFundWeb.AdServeController do
     cond do
       template == nil ->
         templates = Templates.list_templates()
-        template_slugs = Enum.map(templates, fn c -> c.slug end)
+
+        available_templates =
+          Enum.map(templates, fn c -> c.slug end)
+          |> Enum.join("|")
 
         conn
         |> put_status(:not_found)
         |> put_resp_content_type("application/javascript")
         |> text(
-          "console.log('CodeFund template does not exist. Available templates are \\'#{
-            Enum.join(template_slugs, "','")
-          }\\'');"
+          "console.log('CodeFund template does not exist. Available templates are [#{
+            available_templates
+          }]');"
         )
 
       true ->
         theme = Enum.find(template.themes, fn t -> t.slug == theme_slug end)
 
         cond do
           theme == nil ->
-            themes = Themes.list_themes()
+            themes = Themes.list_themes_for_template(template)
             theme_slugs = Enum.map(themes, fn c -> c.slug end)
 
+            available_slugs =
+              theme_slugs
+              |> Enum.uniq()
+              |> Enum.join("|")
+
             conn
             |> put_resp_content_type("application/javascript")
```

Comment:
Do we need to return 404 here too? Not applicable though if we restructure this function.

Reply:
Good find. Yes